llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6743,8 +6743,25 @@ static bool SwitchToLookupTable(SwitchInst *SI, IRBuilder<> &Builder,
     TableSize =
         (MaxCaseVal->getValue() - MinCaseVal->getValue()).getLimitedValue() + 1;
 
+  // If the default destination is unreachable, or if the lookup table covers
+  // all values of the conditional variable, branch directly to the lookup table
+  // BB. Otherwise, check that the condition is within the case range.
+  bool DefaultIsReachable = !SI->defaultDestUndefined();
+
   bool TableHasHoles = (NumResults < TableSize);
-  bool NeedMask = (TableHasHoles && !HasDefaultResults);
+
+  // If the table has holes but the default destination doesn't produce any
+  // constant results, the lookup table entries corresponding to the holes will
+  // contain undefined values.
+  bool HolesHaveUndefinedResults = (TableHasHoles && !HasDefaultResults);
```

Comment:
```suggestion
  bool HolesHaveUndefinedResults = TableHasHoles && !HasDefaultResults;
```

Reply:
My bad, I was keeping with the style of the existing code. I've removed these parens (and the other ones you pointed out)